### ISSUE TITLE
Attacks on Inception_v1 (googlenet)

### ISSUE THREAD

**Issue by rajasekharponakala** [BorealisAI/advertorch#5]
Hi Team. Thanks for the awesome code. Can someone guide me to train googlenet as an adversarially robust model. 

I have trained (pytorch implementation of googlenet) a my model on classification dataset of 18k sample with 5 categories. 

Now, how to attack the model by using the pretrained weights?

**Comment by gwding** [comment]
In principle, for adversarial training you would want to do something similar to this piece of code, https://github.com/BorealisAI/advertorch/blob/master/advertorch_examples/tutorial_train_mnist.py#L69-L73

Maybe your have a more specific question?

**Comment by rajasekharponakala** [comment]
@gwding: Hi, thanks!.

I did training and went well. Now, I want to plot (may be using matplotlib ) the graphs for  validation accuracy/loss Vs. training epochs. 

may be around here: https://github.com/BorealisAI/advertorch/blob/5c2c9248c71479eb54e6511e8a69415661159288/advertorch_examples/tutorial_train_mnist.py#L119

could you give any suggestions or example script for using matplotlib to plot graph?

thanks!.

**Comment by gwding** [comment]
@rajasekharponakala I would suggest following common matplotlib examples for plotting, the epoch-accuracy curve doesn't seem to be special in anyway. if you want to plot these in real time, i suggest you have a look at https://github.com/lanpa/tensorboardX

**Comment by rajasekharponakala** [comment]
Hi!
For Inception_v1, there are two `aux branches`, can we include them at the `train_phase`, to be something like this:
```
output, aux1, aux2 = model(data)
loss1 = criterion(ouputs, targets)
loss2 = criterion(aux1, targets)
loss3 = criterion(aux2, targets)
loss = loss1 + 0.3 * (loss2+loss3)
```
the corresponding repo snippet around here:
https://github.com/BorealisAI/advertorch/blob/5c2c9248c71479eb54e6511e8a69415661159288/advertorch_examples/tutorial_train_mnist.py#L76

could suggest some thoughts?

**Comment by gwding** [comment]
@rajasekharponakala I have no experience of adversarial training on inception models, so I'm not exactly sure. I guess it depends on whether you would use the aux branches at test time. If yes, then it's natural to adversarially train on them. If no, whether (and the weight) to adversarially train on them seems to be a hyperparameter to me.
But I feel this is a very interesting question, if you find out which way is better, I'd be curious to know :)
